bookkeeper-server/src/main/java/org/apache/bookkeeper/util/ReflectionUtils.java

```diff
@@ -36,6 +35,30 @@
             new ConcurrentHashMap<Class<?>, Constructor<?>>();
 
     /**
+     * Returns the {@code Class} object object associated with the class or interface
+     * with the given string name, which is a subclass of {@code xface}.
+     *
+     * @param className class name
+     * @param xface class interface
+     * @return the class object associated with the class or interface with the given string name.
+     */
+    public static <U> Class<? extends U> forName(String className,
+                                                 Class<U> xface) {
+
+        // Construct the class
+        Class<?> theCls;
+        try {
+            theCls = Class.forName(className);
```

Comment:
I think is does not work very well, it is better to use Thread.currentThread().getContextClassloader()
I know that for bookie started from the command line this is not an issue, but if you embed the bookie in another process the class loader will be important.
I see that anyway the same approach is used in another method in this class.
So I am fine with this change, we can improve it later.

Reply:
I don't think we have ever used `Thread.currentThread().getContextClassLoader()`. I think that is typically useful in a multi-classloader environment. I am not sure the bookkeeper usage will ever get into this situation.